tests/test_gene_association_file.py

```diff
@@ -0,0 +1,136 @@
+        {"Evidence_Code": "IDA", "publications": "GO_REF:0000001|PMID:123456"}, # Multiple publications (explode test)
+        {"Evidence_Code": "IMP", "publications": None}, # Null publications
+    ])
+
+# --- Tests ---
+@pytest.mark.unit
+def test_go_annotation_processed(temp_csv_files, evidence_mapping_df):
+    input_path, output_path = temp_csv_files
+    process_go_annotations(str(input_path), str(output_path), evidence_mapping_df)
+    assert output_path.exists(), "Output file not created."
+    df = pd.read_csv(output_path)
+    expected_cols = set(ASSOCIATION_COL_TYPES.keys())
+    assert expected_cols.issubset(set(df.columns)), (
+        f"Output columns do not match expected.\nExpected: {expected_cols}\nGot: {set(df.columns)}"
+    )
+
+@pytest.mark.integration
+def test_io_behavior(temp_csv_files, evidence_mapping_df):
+    """Integration test for I/O behavior of process_go_annotations."""
+    input_path, output_path = temp_csv_files
+    assert not output_path.exists(), "Output file already exist before processing."
+    process_go_annotations(str(input_path), str(output_path), evidence_mapping_df)
+    assert output_path.exists(), "Output file not created."
+    df_out = pd.read_csv(output_path)
+    assert not df_out.empty, "Output DataFrame is empty."
+    for col in ASSOCIATION_COL_TYPES.keys():
+        assert col in df_out.columns, f"Column {col} missing in output."
+
+@pytest.mark.unit
+def test_column_types(temp_csv_files, evidence_mapping_df):
+    input_path, output_path = temp_csv_files
+    process_go_annotations(str(input_path), str(output_path), evidence_mapping_df)
+    df = pd.read_csv(output_path)
+    for column, expected_type in ASSOCIATION_COL_TYPES.items():
+        not_null = df[column].dropna()
+        if not not_null.empty:
+            mismatches = ~not_null.map(lambda x: isinstance(x, expected_type))
+            assert not mismatches.any(), f"Column {column} has incorrect types."
+
+@pytest.mark.unit
+def test_negated_logic(temp_csv_files, evidence_mapping_df):
+    input_path, output_path = temp_csv_files
+    process_go_annotations(str(input_path), str(output_path), evidence_mapping_df)
+    df = pd.read_csv(output_path)
+    assert df["negated"].any(), "Expected at least one negated is True row, it present NOT|enables"
+    assert (~df["negated"]).any(), "Expected at least one negated is False row, it present enables"
+
+# --- Helper function for evidence mapping test --- 
+@pytest.mark.unit
+def test_merge_and_fallback(annotation_df, evidence_mapping_df):
+    """Test the evidence mapping merge and fallback logic."""
+    merged = merge_evidence_mapping(annotation_df, evidence_mapping_df)
+    assert merged.loc[0, "evidence_type"] == "ECO:0000250", "Direct match failed."
+    assert merged.loc[1, "evidence_type"] == "ECO:0000307", "Fallback match failed."
+    assert "DB_Reference" not in merged.columns, "'DB_Reference' should have been dropped after merge."
```

Comment:
just FYI (you don't need to change anything), most of the time I don't bother with the description after the `assert` statement. If, in future tests, you want to omit those strings, that's fine with me!